### PR TITLE
feat(sqlite/query-generator): support restart identity in SQLite

### PR DESCRIPTION
**Pull request by SavageCore**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [x] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Have you added new tests to prevent regressions?
- [ ] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [x] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/master/CONTRIBUTING.md)?

### Description of change

Follow-up to #5356 adding support for SQLite.

`npm run test-sqlite` passes and I also tested the generated SQL against my DB

I believe current docs suffice?

### CODE REVIEW

File: lib/dialects/sqlite/query-generator.js [hunk 1]
```diff
@@ -257,8 +257,11 @@ class SQLiteQueryGenerator extends MySqlQueryGenerator {
     return { query, bind };
   }
 
-  truncateTableQuery(tableName) {
-    return `DELETE FROM ${this.quoteTable(tableName)}`;
+  truncateTableQuery(tableName, options = {}) {
+    return [
+      `DELETE FROM ${this.quoteTable(tableName)}`,
+      options.restartIdentity ? `; DELETE FROM \`sqlite_sequence\` WHERE \`name\` = '${tableName.schema}.${tableName.table}';` : ''
```
Comment (sushantdhiman) [inline lib/dialects/sqlite/query-generator.js]: `this.quoteTable` for `sqlite_sequence`, `this.quoteIdentifier` for `name`

Comment (sushantdhiman) [reply lib/dialects/sqlite/query-generator.js]: `'${tableName.schema}.${tableName.table}'` should be replaced with `this.quoteTable(tableName)`

Comment (SavageCore) [reply lib/dialects/sqlite/query-generator.js]: I tried with quoteTable and it used backticks which wasn't valid, must be single inverted comma?

Comment (sushantdhiman) [reply lib/dialects/sqlite/query-generator.js]: Isn't delete from statement using same `quoteTable`, that works.

Comment (SavageCore) [reply lib/dialects/sqlite/query-generator.js]: From my testing it seemed that 

```
DELETE FROM `table`; DELETE FROM `sqlite_sequence` WHERE `name` = 'table'
```

Works but

```
DELETE FROM `table`; DELETE FROM `sqlite_sequence` WHERE `name` = `table`
```

Returns 

```
-- At line 1:
DELETE FROM `sqlite_sequence` WHERE `name` = `table`
-- Result: no such column: table
```

Comment (sushantdhiman) [reply lib/dialects/sqlite/query-generator.js]: Yeah that wont work as we are comparing against string `name`, you can try `'this.escape(tableName.table)'`. It is possible `tableName` is already a string, so you need to handle that as well

Do you think we need to prefix `schema` with table name? How is `sqlite_sequence` is storing table names, prefixed with schema?

Comment (SavageCore) [reply lib/dialects/sqlite/query-generator.js]: > 
> 
> Yeah that wont work as we are comparing against string `name`, you can try `'this.escape(tableName.table)'`. It is possible `tableName` is already a string, so you need to handle that as well
> 
> Do you think we need to prefix `schema` with table name? How is `sqlite_sequence` is storing table names, prefixed with schema?

Yes, `sqlite_sequence` name is `public.test_users` in the test examples.

File: lib/dialects/sqlite/query-generator.js [hunk 2]
```diff
@@ -257,8 +257,11 @@ class SQLiteQueryGenerator extends MySqlQueryGenerator {
     return { query, bind };
   }
 
-  truncateTableQuery(tableName) {
-    return `DELETE FROM ${this.quoteTable(tableName)}`;
+  truncateTableQuery(tableName, options = {}) {
+    return [
+      `DELETE FROM ${this.quoteTable(tableName)}`,
+      options.restartIdentity ? `; DELETE FROM ${this.quoteTable('sqlite_sequence')} WHERE ${this.quoteIdentifier('name')} = '${tableName.schema}.${tableName.table}';` : ''
```
Comment (sushantdhiman) [inline lib/dialects/sqlite/query-generator.js]:
```suggestion
      options.restartIdentity ? `; DELETE FROM ${this.quoteTable('sqlite_sequence')} WHERE ${this.quoteIdentifier('name')} = '${this.escape(tableName.schema)}.${this.escape(tableName.table)}';` : ''
```

Comment (SavageCore) [reply lib/dialects/sqlite/query-generator.js]: Oh hang on, in both my original and this suggestion we don't handle cases without schema. We need the logic from quoteTable.

Comment (sushantdhiman) [reply lib/dialects/sqlite/query-generator.js]: You can use ```Utils.addTicks(Utils.removeTicks(this.quoteTable(table), '`'), "'");```

**Pull request merged**